### DISCUSSION TITLE
Redirection error Cross-Origin Request Blocked: The Same Origin Policy disallows reading the remote resource at .....

### DISCUSSION THREAD

**Discussion by sraturi** [unpoly/unpoly#316] [Q&A] [answered]
Hello,

I am using unpoly within a quarkus project. We use fragment loading extensively i.e `up-follow`  

If we let the application run for a bit and the SSO has a timeout. We get a redirect request (to sso login)

In the console, I see that we are hitting the `up:request:fatal` and the network tab errors out with a 302 status with error: `Cross-Origin Request Blocked: The Same Origin Policy disallows reading the remote resource at abc.com`

 I want to force a full page reload if we are hitting the 302 status code.

I tried doing something like 
```
        up.on('up:request:fatal',function (event) {
          console.log("I am in fatal")
            //location.reload()

        })
```
but the event, in this case, does not have a response.

**Comment by adam12** [comment, marked as answer]
Does `event` have a `request`? Maybe you can do this:

```javascript
up.on('up:request:fatal', function(event) {
  event.request.loadPage()
})
```

Here's the documentation for [`up.Request#loadPage`](https://unpoly.com/up.Request.prototype.loadPage).

**Reply by sraturi** [reply]
`event` does have a request but ideally, I would like to know if this is in fact a `redirect` rather than just reloading the page on every fatal.

**Reply by adam12** [reply]
I'm not sure it's going to be possible without changes to Unpoly, but maybe someone else has a better idea.

**Reply by triskweline** [reply]
> I would like to know if this is in fact a redirect rather than just reloading the page on every fatal.

I don't think inspecting a CORS failure is possible from JavaScript. There is no browser API to even see a redirect response. The browser will automatically make a second request for the new `Location` and, in this case, fail.

If you're in control of the code doing the SSO redirect, consider signaling the expired session by another means, such as setting an HTTP header that you can inspect in `up:fragment:loaded`.

If you cannot control the redirect, maybe reloading the page for fatal `GET` requests would not be such a bad default.

**Reply by sraturi** [reply]
Thank you, we ended up with page reload on fatal since `fatal` only happens after _timeout or loss of network connectivity_
